From a436dea010f47449f25ebec68d1af52074006496 Mon Sep 17 00:00:00 2001
From: Eric Chanudet <chanudete@ainfosec.com>
Date: Mon, 13 Aug 2018 14:02:47 -0400
Subject: [PATCH 20/22] tboot-utils: Fix tools build in 64bits env.

Current printk() format will not handle 64bit values without triggering
compilation errors.

Signed-off-by: Eric Chanudet <chanudete@ainfosec.com>
---
 include/tboot.h | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/include/tboot.h b/include/tboot.h
index 4e7c7be..ad4c544 100644
--- a/include/tboot.h
+++ b/include/tboot.h
@@ -152,6 +152,15 @@ static inline bool tboot_in_measured_env(void)
     return (g_tboot_shared != NULL);
 }
 
+#if __WORDSIZE == 64
+# define __PRI64_PREFIX	"l"
+# define __PRIPTR_PREFIX	"l"
+#else
+# define __PRI64_PREFIX	"ll"
+# define __PRIPTR_PREFIX
+#endif
+#define PRIu64		__PRI64_PREFIX "u"
+
 static inline void print_tboot_shared(const tboot_shared_t *tboot_shared)
 {
     printk(TBOOT_DETA"tboot_shared data:\n");
@@ -165,8 +174,8 @@ static inline void print_tboot_shared(const tboot_shared_t *tboot_shared)
     printk(TBOOT_DETA"\t flags: 0x%8.8x\n", tboot_shared->flags);
     printk(TBOOT_DETA"\t ap_wake_addr: 0x%08x\n", (uint32_t)tboot_shared->ap_wake_addr);
     printk(TBOOT_DETA"\t ap_wake_trigger: %u\n", tboot_shared->ap_wake_trigger);
-    printk(TBOOT_DETA"\t evt_log_region: 0x%llx\n", tboot_shared->evt_log_region);
-    printk(TBOOT_DETA"\t evt_log_size: 0x%llx\n", tboot_shared->evt_log_size);
+    printk(TBOOT_DETA"\t evt_log_region: 0x%"PRIu64"\n", tboot_shared->evt_log_region);
+    printk(TBOOT_DETA"\t evt_log_size: 0x%"PRIu64"\n", tboot_shared->evt_log_size);
 }
 
 #endif    /* __TBOOT_H__ */
-- 
2.16.1

